### PR TITLE
Fix #540: Added clients section to parse driver name/version along with ip that initiated connection

### LINKED ISSUE #540: Include client information summary in mloginfo
Logs contain a line:

```
NETWORK  [conn1] received client metadata from 127.0.0.1:50426 conn1: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.4.10" }, os: { type: "Darwin", name: "Mac OS X", architecture: "x86_64", version: "16.7.0" } }
```

It would be good to have this OS and client information summarized as an option in mloginfo.

### PR DESCRIPTION
**Pull request by garycahill**
## Description of changes

Added clients section to parse driver name/version along with ip that initiated connection
Driver name and version are stored as keys in dictionary variable with unique ip's added as value in list form.

--help updated also

## Testing

to run:
mloginfo mongodb.log --clients

Tested against multiple log files

O/S testing:
| O/S              | Version(s)
| ---------------- | -----------
| Linux            | Ubuntu 18.04, Centos 7.4
| macOS            | 
| Windows          |

### CODE REVIEW

File: mtools/mloginfo/mloginfo.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(self):
         self.argparser_sectiongroup = self.argparser.add_argument_group(inf,
                                                                         cmds)
 
-        # add all filter classes from the filters module
+        # add all filter classes from the filters moduleL
```
Comment (stennie) [inline mtools/mloginfo/mloginfo.py]: Typo to remove ;)

File: mtools/mloginfo/sections/clients_section.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        for logevent in self.mloginfo.logfile:
+            line = logevent.line_str
+
+
+
+            pos = line.find('client metadata')
+            if pos != -1:
+                tokens = line[pos:pos + 130].split(' ')
+                #handle mgo as driver and verion backwards
```
Comment (stennie) [inline mtools/mloginfo/sections/clients_section.py]: Not clear what this comment is referring to?

File: mtools/mloginfo/sections/clients_section.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        driver_info = {}
+
+        for logevent in self.mloginfo.logfile:
+            line = logevent.line_str
+
+
+
+            pos = line.find('client metadata')
+            if pos != -1:
+                tokens = line[pos:pos + 130].split(' ')
```
Comment (stennie) [inline mtools/mloginfo/sections/clients_section.py]: Please add a comment with context for the 130 magic number. Is that the maximum size of client metadata?

File: mtools/mloginfo/sections/clients_section.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+class ClientSection(BaseSection):
+    """
+    ClientSection class.
+
+    This section goes through the logfile and extracts driver information from
+    connected clients.
+    """
+
+    name = "client driver info"
```
Comment (stennie) [inline mtools/mloginfo/sections/clients_section.py]: The name should match the section (this should be `clients`)

File: mtools/mloginfo/sections/clients_section.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    dv_formatted = str(driver[1:-2])+":"+str(version[1:-1])
+                    if dv_formatted not in driver_info:
+                        driver_info[dv_formatted] = [ip_formatted]
+                    elif dv_formatted in driver_info:
+                        if ip_formatted in driver_info.get(dv_formatted):
+                            continue
+                        else:
+                            driver_info[dv_formatted].append(ip_formatted)
+                elif tokens[9] == '\"MongoDB' and tokens[10] == 'Internal':
+                    driver = 'MongoDB Internal Driver'
```
Comment (stennie) [inline mtools/mloginfo/sections/clients_section.py]: Is there a better description than MongoDB Internal Driver? Does the following token provide any context?

**Comment by stennie** [comment]
Also, can you confirm which versions of Python you tested with? We're currently testing with 2.7.x and 3.6.x.

Thanks,
Stennie

File: mtools/mloginfo/mloginfo.py (hunk 1 above)
Comment (garycahill) [reply mtools/mloginfo/mloginfo.py]: Typo removed

File: mtools/mloginfo/sections/clients_section.py (hunk 2 above)
Comment (garycahill) [reply mtools/mloginfo/sections/clients_section.py]: Added comment to code clarify 
#Occasionally mgo drivers are logged as driver: version -> name, below to handle

File: mtools/mloginfo/sections/clients_section.py (hunk 3 above)
Comment (garycahill) [reply mtools/mloginfo/sections/clients_section.py]: MongoDB Internal Driver was pushing version number outside index, increased to 120 to accommodate (changed from 130)

File: mtools/mloginfo/sections/clients_section.py (hunk 4 above)
Comment (garycahill) [reply mtools/mloginfo/sections/clients_section.py]: Corrected

File: mtools/mloginfo/sections/clients_section.py (hunk 5 above)
Comment (garycahill) [reply mtools/mloginfo/sections/clients_section.py]: Unfortunately not, it is presented in logfile the same as other drivers --> 
{ driver: { name: "MongoDB Internal Client", version: "4.0.4" }

**Comment by niccottrell** [comment]
Looks like the automated checks are failing during to a matplotlib error. Does the test harness need to be updated?

**Pull request merged**